packages/hash/frontend/src/components/hooks/blockProtocolFunctions/knowledge/useBlockProtocolGetEntity.ts

```diff
@@ -82,47 +90,27 @@ export const useBlockProtocolGetEntity = (): {
         };
       }
 
-      const { persistedEntity } = entityResponseData;
-      const { outgoingPersistedLinks } = outgoingLinksData;
-
+      // Manually patch the subgraph to add links to the inner entity
       /**
-       * @todo: obtain this sub-graph as part of the prior `getEntity` query.
-       * May be addressed as part of https://app.asana.com/0/1200211978612931/1203089535761796/f or related work.
+       * @todo: remove this when we start returning links in the subgraph
+       *   https://app.asana.com/0/0/1203214689883095/f
        */
+      const subgraph = entityResponseData.getPersistedEntity;
 
-      const { data: entityTypeResponseData } =
-        await getEntityTypeRootedSubgraphFn({
-          query: getEntityTypeRootedSubgraphQuery,
-          variables: {
-            entityTypeId: persistedEntity.entityTypeId,
-            dataTypeResolveDepth: 255,
-            propertyTypeResolveDepth: 255,
-            linkTypeResolveDepth: 255,
-            entityTypeResolveDepth: 1,
-          },
-        });
-
-      if (!entityTypeResponseData) {
-        return {
-          errors: [
-            {
-              code: "INVALID_INPUT",
-              message:
-                "Error getting the subgraph rooted at entity's entity type",
-            },
-          ],
-        };
+      for (const [vertexId, vertex] of Object.entries(subgraph.vertices)) {
+        if (vertex.kind === "entity") {
+          (vertex.inner as unknown as Entity).links =
```

Comment:
A re-usable type-guard would be better than having to cast every time, so if these typing limitations aren't temporary I think we should add one.

Reply:
There is an `isEntityVertex` type guard but it uses the patched version of the `Subgraph` type. So a type guard here wouldn't actually be re-usable because it would be defined for the unpatched version (and this will only happen here).

These typing limitations are temporary, when we fix the GraphQL types we won't need any of the patching.

Reply:
> when we fix the GraphQL types we won't need any of the patching

Okay, maybe let's leave a todo if these castings are temporary then?

Reply:
There is a todo right above it

Reply:
Cool thanks, missed that it was related to these lines